### ISSUE TITLE
meta-dump without arguments just gives me AssertionError

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad-metalad#158]
I guess if there is a mandatory argument it should be `InsufficientArgumentsError` and ideally `--help` summary printed .

ATM

```
$> datalad meta-dump 
[ERROR  ] AssertionError() (AssertionError) 
```

<details><summary>DataLad 0.15.2 WTF (configuration, credentials, datalad, dataset, dependencies, environment, extensions, git-annex, location, metadata_extractors, metadata_indexers, python, system)</summary>

# WTF
## credentials 
  - keyring: 
    - active_backends: 
      - SecretService Keyring
      - PlaintextKeyring with no encyption v.1.0 at /home/yoh/.local/share/python_keyring/keyring_pass.cfg
    - config_file: /home/yoh/.config/python_keyring/keyringrc.cfg
    - data_root: /home/yoh/.local/share/python_keyring
## datalad 
  - full_version: 0.15.2
  - version: 0.15.2
## dataset 
  - branches: 
    - git-annex@7864cb5
    - master@c090537
  - id: 9040fae2-8704-11e8-aff3-0242ac12000c
  - metadata: <SENSITIVE, report disabled by configuration>
  - path: /home/yoh/datalad/openneuro/ds000003
  - repo: AnnexRepo
## dependencies 
  - annexremote: 1.5.0
  - appdirs: 1.4.4
  - boto: 2.49.0
  - cmd:7z: 16.02
  - cmd:annex: 8.20210903+git120-gdb55bec8b-1~ndall+1
  - cmd:bundled-git: 2.24.0
  - cmd:git: 2.24.0
  - cmd:system-git: 2.33.0
  - cmd:system-ssh: 8.4p1
  - humanize: 3.11.0
  - iso8601: 0.1.16
  - keyring: 23.2.1
  - keyrings.alt: 4.1.0
  - msgpack: 1.0.2
  - requests: 2.26.0
  - wrapt: 1.12.1
## environment 
  - GIT_PAGER: less --no-init --quit-if-one-screen
  - LANG: en_US.UTF-8
  - PATH: /home/yoh/proj/datalad/datalad-fuse/venvs/dev3/bin:/home/yoh/gocode/bin:/home/yoh/gocode/bin:/home/yoh/bin:/home/yoh/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/sbin:/usr/sbin:/usr/local/sbin
## extensions 
  - deprecated: 
    - description: Deprecated functionality
    - entrypoints: 
      - datalad_deprecated.ls.Ls: 
        - class: Ls
        - load_error: None
        - module: datalad_deprecated.ls
        - names: 
      - datalad_deprecated.publish.Publish: 
        - class: Publish
        - load_error: None
        - module: datalad_deprecated.publish
        - names: 
    - load_error: None
    - module: datalad_deprecated
    - version: 0.1+43.g938101d
  - fuse: 
    - description: DataLad FUSE command suite
    - entrypoints: 
      - datalad_fuse.FuseFS: 
        - class: FuseFS
        - load_error: None
        - module: datalad_fuse
        - names: 
          - fusefs
          - fusefs
      - datalad_fuse.fsspec_cache_clear.FsspecCacheClear: 
        - class: FsspecCacheClear
        - load_error: None
        - module: datalad_fuse.fsspec_cache_clear
        - names: 
          - fsspec-cache-clear
          - fsspec_cache_clear
      - datalad_fuse.fsspec_head.FsspecHead: 
        - class: FsspecHead
        - load_error: None
        - module: datalad_fuse.fsspec_head
        - names: 
          - fsspec-head
          - fsspec_head
    - load_error: None
    - module: datalad_fuse
    - version: 0+untagged.62.g1098dae
  - metalad: 
    - description: DataLad semantic metadata command suite
    - entrypoints: 
      - datalad_metalad.add.Add: 
        - class: Add
        - load_error: None
        - module: datalad_metalad.add
        - names: 
          - meta-add
          - meta_add
      - datalad_metalad.aggregate.Aggregate: 
        - class: Aggregate
        - load_error: None
        - module: datalad_metalad.aggregate
        - names: 
          - meta-aggregate
          - meta_aggregate
      - datalad_metalad.conduct.Conduct: 
        - class: Conduct
        - load_error: None
        - module: datalad_metalad.conduct
        - names: 
          - meta-conduct
          - meta_conduct
      - datalad_metalad.dump.Dump: 
        - class: Dump
        - load_error: None
        - module: datalad_metalad.dump
        - names: 
          - meta-dump
          - meta_dump
      - datalad_metalad.extract.Extract: 
        - class: Extract
        - load_error: None
        - module: datalad_metalad.extract
        - names: 
          - meta-extract
          - meta_extract
    - load_error: None
    - module: datalad_metalad
    - version: 0.3.0a1+50.g230e1d3
  - neuroimaging: 
    - description: Neuroimaging tools
    - entrypoints: 
      - datalad_neuroimaging.bids2scidata.BIDS2Scidata: 
        - class: BIDS2Scidata
        - load_error: None
        - module: datalad_neuroimaging.bids2scidata
        - names: 
          - bids2scidata
    - load_error: None
    - module: datalad_neuroimaging
    - version: 0.3.1
## git-annex 
  - build flags: 
    - Assistant
    - Webapp
    - Pairing
    - Inotify
    - DBus
    - DesktopNotify
    - TorrentParser
    - MagicMime
    - Feeds
    - Testsuite
    - S3
    - WebDAV
  - dependency versions: 
    - aws-0.20
    - bloomfilter-2.0.1.0
    - cryptonite-0.25
    - DAV-1.3.3
    - feed-1.0.0.0
    - ghc-8.4.4
    - http-client-0.5.13.1
    - persistent-sqlite-2.8.2
    - torrent-10000.1.1
    - uuid-1.3.13
    - yesod-1.6.0
  - key/value backends: 
    - SHA256E
    - SHA256
    - SHA512E
    - SHA512
    - SHA224E
    - SHA224
    - SHA384E
    - SHA384
    - SHA3_256E
    - SHA3_256
    - SHA3_512E
    - SHA3_512
    - SHA3_224E
    - SHA3_224
    - SHA3_384E
    - SHA3_384
    - SKEIN256E
    - SKEIN256
    - SKEIN512E
    - SKEIN512
    - BLAKE2B256E
    - BLAKE2B256
    - BLAKE2B512E
    - BLAKE2B512
    - BLAKE2B160E
    - BLAKE2B160
    - BLAKE2B224E
    - BLAKE2B224
    - BLAKE2B384E
    - BLAKE2B384
    - BLAKE2BP512E
    - BLAKE2BP512
    - BLAKE2S256E
    - BLAKE2S256
    - BLAKE2S160E
    - BLAKE2S160
    - BLAKE2S224E
    - BLAKE2S224
    - BLAKE2SP256E
    - BLAKE2SP256
    - BLAKE2SP224E
    - BLAKE2SP224
    - SHA1E
    - SHA1
    - MD5E
    - MD5
    - WORM
    - URL
    - X*
  - local repository version: 8
  - operating system: linux x86_64
  - remote types: 
    - git
    - gcrypt
    - p2p
    - S3
    - bup
    - directory
    - rsync
    - web
    - bittorrent
    - webdav
    - adb
    - tahoe
    - glacier
    - ddar
    - git-lfs
    - httpalso
    - borg
    - hook
    - external
  - supported repository versions: 
    - 8
  - upgrade supported from repository versions: 
    - 0
    - 1
    - 2
    - 3
    - 4
    - 5
    - 6
    - 7
  - version: 8.20210903+git120-gdb55bec8b-1~ndall+1
## location 
  - path: /home/yoh/datalad/openneuro/ds000003
  - type: dataset
## metadata_extractors 
  - annex (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: None
    - module: datalad.metadata.extractors.annex
    - version: None
  - audio (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: ModuleNotFoundError(No module named 'mutagen')
    - module: datalad.metadata.extractors.audio
  - bids (datalad-neuroimaging 0.3.1): 
    - distribution: datalad-neuroimaging 0.3.1
    - load_error: None
    - module: datalad_neuroimaging.extractors.bids
    - version: None
  - datacite (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: None
    - module: datalad.metadata.extractors.datacite
    - version: None
  - datalad_core (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: None
    - module: datalad.metadata.extractors.datalad_core
    - version: None
  - datalad_rfc822 (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: None
    - module: datalad.metadata.extractors.datalad_rfc822
    - version: None
  - dicom (datalad-neuroimaging 0.3.1): 
    - distribution: datalad-neuroimaging 0.3.1
    - load_error: None
    - module: datalad_neuroimaging.extractors.dicom
    - version: None
  - exif (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: ModuleNotFoundError(No module named 'exifread')
    - module: datalad.metadata.extractors.exif
  - external_dataset (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.external_dataset
    - version: None
  - external_file (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.external_file
    - version: None
  - frictionless_datapackage (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: None
    - module: datalad.metadata.extractors.frictionless_datapackage
    - version: None
  - image (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: ModuleNotFoundError(No module named 'PIL')
    - module: datalad.metadata.extractors.image
  - metalad_annex (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.annex
    - version: None
  - metalad_core (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.core
    - version: None
  - metalad_core_dataset (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.core_dataset
    - version: None
  - metalad_core_file (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.core_file
    - version: None
  - metalad_custom (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.custom
    - version: None
  - metalad_external_dataset (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.external_dataset
    - version: None
  - metalad_external_file (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.external_file
    - version: None
  - metalad_runprov (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.runprov
    - version: None
  - metalad_studyminimeta (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.extractors.studyminimeta.main
    - version: None
  - nidm (datalad-neuroimaging 0.3.1): 
    - distribution: datalad-neuroimaging 0.3.1
    - load_error: None
    - module: datalad_neuroimaging.extractors.nidm
    - version: None
  - nifti1 (datalad-neuroimaging 0.3.1): 
    - distribution: datalad-neuroimaging 0.3.1
    - load_error: None
    - module: datalad_neuroimaging.extractors.nifti1
    - version: None
  - xmp (datalad 0.15.2): 
    - distribution: datalad 0.15.2
    - load_error: ModuleNotFoundError(No module named 'libxmp')
    - module: datalad.metadata.extractors.xmp
## metadata_indexers 
  - metalad_studyminimeta (datalad-metalad 0.3.0a1): 
    - distribution: datalad-metalad 0.3.0a1
    - load_error: None
    - module: datalad_metalad.indexers.studyminimeta
    - version: None
## python 
  - implementation: CPython
  - version: 3.9.7
## system 
  - distribution: debian/11/bullseye
  - encoding: 
    - default: utf-8
    - filesystem: utf-8
    - locale.prefered: UTF-8
  - max_path_length: 292
  - name: Linux
  - release: 5.10.0-7-amd64
  - type: posix
  - version: #1 SMP Debian 5.10.40-1 (2021-05-28)
</details>

**Comment by christian-monch** [comment]
> so the idea there to make it work on all known to dataset files if no expression is provided?

If no pattern is provided it defaults to `"."`. That means it will list root-metadata entries of the dataset. (The dataset is either given by the current directory or by the `-d`, `--dataset` argument.)

**Comment by christian-monch** [comment]
Fixed by PR #164
